src/SelfServe/SqlX/SqlX.tsx

```diff
@@ -6,41 +6,60 @@ import {
   RefreshResult,
   SelfServeBaseClass,
   SelfServeNotification,
+  SelfServeNotificationType,
   SmartUiInput,
 } from "../SelfServeTypes";
-import { refreshDedicatedGatewayProvisioning } from "./SqlX.rp";
+import {
+  ResourceStatus,
+  refreshDedicatedGatewayProvisioning,
+  updateDedicatedGatewayResource,
+  deleteDedicatedGatewayResource,
+  getCurrentProvisioningState,
+} from "./SqlX.rp";
+
+let disableAttributesOnDedicatedGatewayChange = false;
 
 const onEnableDedicatedGatewayChange = (
   currentState: Map<string, SmartUiInput>,
   newValue: InputType
 ): Map<string, SmartUiInput> => {
   const sku = currentState.get("sku");
   const instances = currentState.get("instances");
-  const isSkuHidden = newValue === undefined || !(newValue as boolean);
+  const hideAttributes = newValue === undefined || !(newValue as boolean);
   currentState.set("enableDedicatedGateway", { value: newValue });
-  currentState.set("sku", { value: sku.value, hidden: isSkuHidden });
-  currentState.set("instances", { value: instances.value, hidden: isSkuHidden });
+  currentState.set("sku", {
+    value: sku.value,
+    hidden: hideAttributes,
+    disabled: disableAttributesOnDedicatedGatewayChange,
+  });
+  currentState.set("instances", {
+    value: instances.value,
+    hidden: hideAttributes,
+    disabled: disableAttributesOnDedicatedGatewayChange,
+  });
   return currentState;
 };
 
+const skuDropDownItems: ChoiceItem[] = [
+  { label: "CosmosD4s", key: "Cosmos.D4s" },
```

Comment:
any reason key and label are different?

Reply:
They are the same, the label ensures that the drop down does not show up as SqlX.CosmosD4s

Reply:
does renaming the label key as "Cosmos.D4s" and changing the translation key create a error? If not, I'd suggest that you keep both the label and key as "CosmosD4s"